src/env.cc

```diff
@@ -917,6 +917,21 @@ void AsyncHooks::grow_async_ids_stack() {
 uv_key_t Environment::thread_local_env = {};
 
 void Environment::Exit(int exit_code) {
+  if (options()->trace_exit) {
+    HandleScope handle_scope(isolate());
+
+    if (is_main_thread()) {
+      fprintf(stderr, "(node:%d) ", uv_os_getpid());
+    } else {
+      fprintf(stderr, "(node:%d, thread:%llu) ", uv_os_getpid(), thread_id());
+    }
+
+    fprintf(
+        stderr, "WARNING: Exited the environment with code %d\n", exit_code);
+    PrintStackTrace(
+        isolate(),
+        StackTrace::CurrentStackTrace(isolate(), 10, StackTrace::kDetailed));
```

Comment:
Is the value 10 from `Error.stackTraceLimit`? How about reusing that instead of using literal value?

Reply:
Opened another PR to address this issue: https://github.com/nodejs/node/pull/30752.